spring-modulith-events/spring-modulith-events-mongodb/src/main/java/org/springframework/modulith/events/mongodb/MongoDbEventPublication.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.modulith.events.mongodb;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.time.Instant;
+
+/**
+ * A MongoDB Document to represent event publications.
+ *
+ * @author Dmitry Belyaev
+ * @author Björn Kieling
+ */
+@Document(collection = "org_springframework_modulith_events")
+@Getter
+@Builder
```

Comment:
Can we remove this one and rather use a constructor directly? It's three properties for manual creation. Spring Data will happily populate the all-args constructor anyway. This also means we can remove the no-arg constructor and rather switch to final fields for everything but the completion date. That one in turn shouldn't need the setter either, should it?

Reply:
We changed the code to use a RequiredArgsConstructor with 3 parameters and 2 package private setters, please review.

Reply:
Additionally package private AllArgsConstructor is provided to be used by Spring Data